packages/calypso-e2e/src/lib/blocks/block-flows/twitter.ts

```diff
@@ -56,10 +57,16 @@ export class TwitterBlockFlow implements BlockFlow {
 	 * @param context The current context for the published post at the point of test execution
 	 */
 	async validateAfterPublish( context: PublishedPostContext ): Promise< void > {
-		const expectedTweetLocator = context.page
+		// In most cases, the iframe will render, so look for that.
+		const iframeTweetLocator = context.page
 			.frameLocator( selectors.publishedTwitterIframe )
 			.locator( `text=${ this.configurationData.expectedTweetText }` )
 			.first(); // May not be specific enough to match only one (and that's okay).
-		await expectedTweetLocator.waitFor();
+
+		// However, sometimes only the bare link renders, so we need to check for that fallback.
+		const bareTweetLinkLocator = context.page.locator( selectors.publishedTwitterBareLink ).first();
+
+		const tweetLocator = iframeTweetLocator.or( bareTweetLinkLocator );
```

Comment:
I wonder if we should throw a `.first()` on here too, just in case.

Reply:
Removed it from the earlier locators and put it on the final one: 03ff6a6bb3